### PR TITLE
Fix webhook lookup

### PR DESCRIPTION
**Pull request by skamoen**
We had a single method for order status updates. However, we used both public and provider references. Oops.

This PR separates lookup by ProviderReference and PublicReference

### CODE REVIEW

File: src/main/java/ch/wisv/payments/rest/MolliePaymentService.java [hunk 1]
```diff
@@ -89,10 +89,20 @@ private void updateOrder(Order order, ResponseOrError<Payment> molliePayment) {
     }
 
     @Override
-    public Order updateStatus(String orderReference) {
-        Order order = orderRepository.findByPublicReference(orderReference)
-                .orElseThrow(() -> new RuntimeException("Order with providerReference " + orderReference + " not found"));
+    public Order updateStatusByPublicReference(String publicOrderReference) {
+        Order order = orderRepository.findByProviderReference(publicOrderReference)
```
Comment (TimvdLippe) [inline src/main/java/ch/wisv/payments/rest/MolliePaymentService.java]: This still is `findByProviderReference`?

Comment (skamoen) [reply src/main/java/ch/wisv/payments/rest/MolliePaymentService.java]: image

**Comment by TimvdLippe** [comment]
I think this is working now? At least I succesfully got an e-mail.

**Comment by skamoen** [comment]
You got the email for another reason, but I still think this works. If changes look good, please merge.

**Review by TimvdLippe — APPROVED** [review]
This should work. Lets see if Martijn can order a ticket.

**Pull request merged**